AsyncSearchActionIT.testRetryVersionConflict fails

**Build scan**:

https://gradle-enterprise.elastic.co/s/xbsokeqogmz4e/console-log?task=:x-pack:plugin:async-search:internalClusterTest

**Repro line**:

`REPRODUCE WITH: ./gradlew ':x-pack:plugin:async-search:internalClusterTest' --tests "org.elasticsearch.xpack.search.AsyncSearchActionIT.testRetryVersionConflict" -Dtests.seed=69CB029E447B660 -Dtests.security.manager=true -Dtests.locale=be -Dtests.timezone=Asia/Kabul -Druntime.java=11`

**Reproduces locally?**:

Not after a handful of iterations

**Applicable branches**:

`master`

**Failure history**:

[Three failures](https://build-stats.elastic.co/app/kibana#/discover?_g=(refreshInterval:(pause:!t,value:0),time:(from:now-90d,mode:quick,to:now))&_a=(columns:!(build.branch),index:b646ed00-7efc-11e8-bf69-63c8ef516157,interval:auto,query:(language:lucene,query:testRetryVersionConflict),sort:!(process.time-start,desc))) on `master` over the last couple of days.

This test was added recently in #63652

**Failure excerpt**:


```

org.elasticsearch.xpack.search.AsyncSearchActionIT > testRetryVersionConflict FAILED |  
-- | --
  | java.lang.AssertionError: [java.util.concurrent.ExecutionException: org.elasticsearch.transport.RemoteTransportException: [node_s0][127.0.0.1:43365][indices:data/write/update[s]]] |  
  | at __randomizedtesting.SeedInfo.seed([69CB029E447B660:1F5DBEBF72D58D61]:0) |  
  | at org.junit.Assert.fail(Assert.java:88) |  
  | at org.junit.Assert.assertTrue(Assert.java:41) |  
  | at org.elasticsearch.xpack.search.AsyncSearchActionIT.testRetryVersionConflict(AsyncSearchActionIT.java:456) |  
```

Also failing on 7.9 https://gradle-enterprise.elastic.co/s/xvofuzl46ajb6

This one contains the nested exception:
```
java.lang.AssertionError: [java.util.concurrent.ExecutionException: RemoteTransportException[[node_s1][127.0.0.1:46393][indices:data/read/async_search/get]]; 
nested: RemoteTransportException[[node_s1][127.0.0.1:46393][indices:data/write/update[s]]]; 
nested: VersionConflictEngineException[[4IS9k-ppQv-tZUSVK5duPQ]: version conflict, required seqNo [22], primary term [1]. current document has seqNo [23] and primary term [1]];]
	at __randomizedtesting.SeedInfo.seed([1A6E99799614A613:3AF97EF00869D12]:0)
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.elasticsearch.xpack.search.AsyncSearchActionIT.testRetryVersionConflict(AsyncSearchActionIT.java:451)
```

Another one on 7.x: https://gradle-enterprise.elastic.co/s/e4mdifcbsntqq

Not reproducible locally.

The exception is similar to above:

```
2> java.lang.AssertionError: [java.util.concurrent.ExecutionException: RemoteTransportException[[node_s1][127.0.0.1:39801][indices:data/read/async_search/get]]; nested: RemoteTransportException[[node_s0][127.0.0.1:42103][indices:data/read/async_search/get]]; nested: RemoteTransportException[[node_s1][127.0.0.1:39801][indices:data/write/update[s]]]; nested: VersionConflictEngineException[[ia_wQNGPRKGbkM-jlsfMAA]: version conflict, required seqNo [17], primary term [1]. current document has seqNo [18] and primary term [1]];] |  
-- | --
  | at __randomizedtesting.SeedInfo.seed([5DE656C6DFF469C8:44275850496652C9]:0) |  
  | at org.junit.Assert.fail(Assert.java:88) |  
  | at org.junit.Assert.assertTrue(Assert.java:41) |  
  | at org.elasticsearch.xpack.search.AsyncSearchActionIT.testRetryVersionConflict(AsyncSearchActionIT.java:453)
```

Another failure on master PR: https://gradle-enterprise.elastic.co/s/3feqp5dosgpwo

Doesn't reproduce for me.

```
./gradlew ':x-pack:plugin:async-search:internalClusterTest' --tests "org.elasticsearch.xpack.search.AsyncSearchActionIT.testRetryVersionConflict" -Dtests.seed=ECA507FC331A242 -Dtests.security.manager=true -Dtests.locale=es-DO -Dtests.timezone=America/Juneau -Druntime.java=11
```


Caused by: org.elasticsearch.index.engine.VersionConflictEngineException: [GYbwbtUpQwaVs1VHs0QcvA]: version conflict, required seqNo [18], primary term [1]. current document has seqNo [19] and primary term [1]

--- a/x-pack/plugin/async-search/src/internalClusterTest/java/org/elasticsearch/xpack/search/AsyncSearchActionIT.java
+++ b/x-pack/plugin/async-search/src/internalClusterTest/java/org/elasticsearch/xpack/search/AsyncSearchActionIT.java
@@ -474,38 +474,4 @@ public class AsyncSearchActionIT extends AsyncSearchIntegTestCase {
         assertNotNull(response.getFailure());
         ensureTaskNotRunning(response.getId());
     }
-
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/63948")
-    public void testRetryVersionConflict() throws Exception {
-        SubmitAsyncSearchRequest request = new SubmitAsyncSearchRequest(indexName);
-        request.setWaitForCompletionTimeout(TimeValue.timeValueMinutes(10));
-        request.setKeepOnCompletion(true);
-        AsyncSearchResponse response = submitAsyncSearch(request);
-        assertNotNull(response.getSearchResponse());
-        assertFalse(response.isRunning());
-
-        List<Thread> threads = new ArrayList<>();
-        CountDownLatch latch = new CountDownLatch(1);
-        List<Exception> exceptions = Collections.synchronizedList(new ArrayList<>());
-        for (int i = 0; i < 2; i++) {
-            Runnable runnable = () -> {
-                for (int j = 0; j < 10; j++) {
-                    try {
-                        latch.await();
-                        getAsyncSearch(response.getId(), TimeValue.timeValueMinutes(10));
-                    } catch (Exception exc) {
-                        exceptions.add(exc);
-                    }
-                }
-            };
-            Thread thread = new Thread(runnable);
-            thread.start();
-            threads.add(thread);
-        }
-        latch.countDown();
-        for (Thread thread : threads) {
-            thread.join();
-        }
-        assertTrue(exceptions.toString(), exceptions.isEmpty());
-    }
 }

--- a/x-pack/plugin/async-search/src/internalClusterTest/java/org/elasticsearch/xpack/search/AsyncSearchActionIT.java
+++ b/x-pack/plugin/async-search/src/internalClusterTest/java/org/elasticsearch/xpack/search/AsyncSearchActionIT.java
@@ -26,13 +26,11 @@ import org.elasticsearch.xpack.core.search.action.AsyncStatusResponse;
 import org.elasticsearch.xpack.core.search.action.SubmitAsyncSearchRequest;
 
 import java.util.ArrayList;
-import java.util.Collections;
 import java.util.HashMap;
 import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
-import java.util.concurrent.CountDownLatch;
 import java.util.concurrent.ExecutionException;
 import java.util.concurrent.atomic.AtomicInteger;
 

